File: 704-binary-search/704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int h(nums.size() - 1), l(0); 
        while(l <= h)
        {
            int mid = h+l / 2;
            if(nums[mid] == target) 
            {
                return mid;
            }
            else if(nums[mid] < target) 
            {
                l = mid + 1;
            }
            else
            {
                h = mid - 1;
            }
        }
        return -1;
    }
};

File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        while start < end:
            mid = start + (end - start) // 2
            if nums[mid] < target:
                start = mid + 1
            else:
                end = mid 
        return start if nums[start] == target else -1